### PR TITLE
feat: override filter status component

### PR DESCRIPTION
**Pull request by leangseu-edx**
[AU-534](https://openedx.atlassian.net/browse/AU-534)

filter

### CODE REVIEW

File: src/containers/ListView/FilterStatusComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const filterNames = state.filters.map((filter) => headerMap[filter.id]);
+  const filterTexts = <p>Filtered by {filterNames.join(', ')}</p>;
+  return (
+    <div className={className}>
+      {showFilteredFields && filterTexts}
+      <Button
+        className={buttonClassName}
+        variant={variant}
+        size={size}
+        onClick={() => setAllFilters([])}
```
Comment (muselesscreator) [inline src/containers/ListView/FilterStatusComponent.jsx]: this needs to be mock-able.

File: src/containers/ListView/FilterStatusComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      >
+        {clearFiltersText}
+      </Button>
+    </div>
+  );
+};
+
+FilterStatusComponent.defaultProps = {
+  className: null,
```
Comment (muselesscreator) [inline src/containers/ListView/FilterStatusComponent.jsx]: because you are passing this, and it shows up in the snapshot, I would pass an empty string

**Review by muselesscreator — CHANGES_REQUESTED** [review]
a couple of testing concerns

File: src/containers/ListView/FilterStatusComponent.jsx (hunk 1 above)
Comment (leangseu-edx) [reply src/containers/ListView/FilterStatusComponent.jsx]: `setAllFilters` is a function that came from the Context, I am not sure how to mock this properly even with class component. Please look at current way I mock it.

File: src/containers/ListView/FilterStatusComponent.test.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      },
+      {
+        id: 'dummy-column',
+        Header: 'Not showing column',
+      },
+    ],
+  };
+  describe('snapshot', () => {
+    beforeEach(() => {
+      jest.spyOn(React, 'useCallback').mockImplementationOnce(() => jest.fn().mockName('clearFilters'));
```
Comment (leangseu-edx) [inline src/containers/ListView/FilterStatusComponent.test.jsx]: This has to be in the same order of `useCallback` being written. It is not the most ideal but there isn't better alternative.

File: src/containers/ListView/FilterStatusComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, DataTableContext } from '@edx/paragon';
+
+const FilterStatusComponent = ({
```
Comment (muselesscreator) [inline src/containers/ListView/FilterStatusComponent.jsx]: suggestion:
define hook method outside the component that grabs the datatable context and returns what is needed.
```
import * as module from './FilterStatusComponent';

const filterHooks = () => {
    const { state, setAllFilters, headers } = React.useContext(DataTableContext);
    if (!setAllFilters || !state.filters) {
      return {};
    }
    const clearFilters = React.useCallback(() => setAllFilters([]), []);
    const headerMap = headers.reduce((obj, cur) => ({ ...obj, [cur.id]: cur.Header }), {});
    const filterNames = state.filters.map(filter => headerMap[filter.id]);
    return { clearFilters, filterNames };
};

const FilterStatusComponents = ({ ... }) => {
  const { clearFilters, filterNames } = module.filterHooks();
  if (!clearFilters {
    return null;
  }
  ...
```

Comment (muselesscreator) [reply src/containers/ListView/FilterStatusComponent.jsx]: then, you can mock the hook method before it goes into the component to verify the 2 pieces you want in the component test, but you can also mock out useCallback to return the callback without calling it, to verify the state before and after it is called (verifying that setAllFilters is called with empty list)

File: src/containers/ListView/FilterStatusComponent.test.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      el.find('Button').simulate('click');
+      expect(mockFilterHooks.clearFilters).toHaveBeenCalledTimes(1);
+    });
+
+    test('on clear filter, set all filter get called with empty array', () => {
+      jest.spyOn(React, 'useContext').mockImplementationOnce(() => ({ ...contextProps }));
+      jest.spyOn(React, 'useCallback').mockImplementationOnce(() => mockSetAllFilters);
+      const el = shallow(<FilterStatusComponent {...props} />);
+      el.find('Button').simulate('click');
+      expect(mockSetAllFilters).toHaveBeenCalledTimes(1);
```
Comment (muselesscreator) [inline src/containers/ListView/FilterStatusComponent.test.jsx:77]: The goal with this test was to verify what the callback actually does.
so, useCallback should be taking a callback and set of memoization pre-reqs, and returning them.  Then we test that the cb actually calls the setAllFilters passed in the contextProps with an empty list.

File: src/containers/ListView/FilterStatusComponent.test.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    test('clear filter click', () => {
+      jest.spyOn(module, 'filterHooks').mockImplementationOnce(() => mockFilterHooks);
+      const el = shallow(<FilterStatusComponent {...props} />);
+      el.find('Button').simulate('click');
+      expect(mockFilterHooks.clearFilters).toHaveBeenCalledTimes(1);
+    });
+
+    test('on clear filter, set all filter get called with empty array', () => {
+      jest.spyOn(React, 'useContext').mockImplementationOnce(() => ({ ...contextProps }));
+      jest.spyOn(React, 'useCallback').mockImplementationOnce(() => mockSetAllFilters);
```
Comment (muselesscreator) [inline src/containers/ListView/FilterStatusComponent.test.jsx]:
```suggestion
      jest.spyOn(React, 'useCallback').mockImplementationOnce((cb, prereqs) => ([cb, prereqs]));
```

**Review by muselesscreator — APPROVED** [review]
Have some testing issues, but I will address those in a follow-on PR.  Hook components should be unit-tested in the same way we test class components: verify behavior, and then verify linkage in snapshots, not by simulating click behavior and checking outcomes.

**Pull request merged**